packages/api/src/brevo-contact/validator/redirect-url.validator.ts

```diff
@@ -22,14 +23,22 @@ export const IsValidRedirectURL = (validationOptions?: ValidationOptions) => {
 export class IsValidRedirectURLConstraint implements ValidatorConstraintInterface {
     constructor(@Inject(BREVO_MODULE_CONFIG) private readonly config: BrevoModuleConfig) {}
 
-    async validate(urlToValidate: string): Promise<boolean> {
-        if (urlToValidate?.startsWith(this.config.brevo.allowedRedirectUrl)) {
-            return true;
+    async validate(urlToValidate: string, args: ValidationArguments): Promise<boolean> {
+        const [scope] = args.constraints;
+
+        try {
+            const configForScope = this.config.brevo.resolveConfig(scope);
+            if (configForScope && urlToValidate?.startsWith(configForScope.allowedRedirectUrl)) {
+                return true;
+            }
+        } catch (error) {
+            console.error(`Error resolving config for scope: ${scope}`, error);
         }
```

Comment:
Why is there a try-catch block here? Which part could throw an error?

In the case of an unsupported scope, wouldn't it be better to throw an error?